federatedscope/core/trainers/trainer.py

```diff
@@ -370,15 +370,15 @@ def update(self, model_parameters):
         self.ctx.model.load_state_dict(self._param_filter(model_parameters),
                                        strict=False)
 
-    def evaluate(self, target_data_split_name="test"):
+    def evaluate(self, mode, target_data_split_name="test"):
         with torch.no_grad():
-            super().evaluate(target_data_split_name)
+            super().evaluate(mode, target_data_split_name)
 
         return self.ctx.eval_metrics
 
-    def validate(self, target_data_split_name="val"):
+    def validate(self, mode, target_data_split_name="val"):
```

Comment:
Have we unified these two interfaces (evaluate, validate) in an early PR?

Reply:
> 
LGTM.

Minor issues:
- We'd better add these distinguishment principle in the documentation of ``context``
- specific the dataset -> specify the dataset